lib/daru/category.rb

```diff
@@ -0,0 +1,883 @@
+    #   dv.to_ints
+    #   # => [0, 1, 0, 1, 2]
+    def to_ints
+      @array
+    end
+
+    # Over rides original inspect for pretty printing in irb
+    def inspect spacing=20, threshold=15
+      row_headers = index.is_a?(MultiIndex) ? index.sparse_tuples : index.to_a
+
+      "#<#{self.class}(#{size})#{metadata && !metadata.empty? ? metadata.inspect : ''}>\n" +
+        Formatters::Table.format(
+          to_a.lazy.map { |v| [v] },
+          headers: @name && [@name],
+          row_headers: row_headers,
+          threshold: threshold,
+          spacing: spacing
+        )
+    end
+
+    # Convert to html for iruby
+    def to_html threshold=30
+      path =
+        if index.is_a?(MultiIndex)
+          File.expand_path('../iruby/templates/vector_mi.html.erb', __FILE__)
+        else
+          File.expand_path('../iruby/templates/vector.html.erb', __FILE__)
+        end
+      ERB.new(File.read(path).strip).result(binding)
+    end
+
+    def to_s
+      to_html
+    end
+
+    # Reorder the vector with given positions
+    # @note Unlike #reindex! which takes index as input, it takes
+    #   positions as an input to reorder the vector
+    # @param [Array] order the order to reorder the vector with
+    # @return reordered vector
+    # @example
+    #   dv = Daru::Vector.new [3, 2, 1], index: ['c', 'b', 'a'], type: :category
+    #   dv.reorder! [2, 1, 0]
+    #   # => #<Daru::Vector(3)>
+    #   #   a   1
+    #   #   b   2
+    #   #   c   3
+    def reorder! order
+      raise ArgumentError, 'Invalid order specified' unless
+        order.sort == size.times.to_a
+      # TODO: Room for optimization
+      old_data = to_a
+      new_data = order.map { |i| old_data[i] }
+      initialize_core_attributes new_data
+      self
+    end
+
+    # Sets new index for vector. Preserves index->value correspondence.
+    # @note Unlike #reorder! which takes positions as input it takes
+    #   index as an input to reorder the vector
+    # @param [Daru::Index, Daru::MultiIndex, Array] idx new index to order with
+    # @return [Daru::Vector] vector reindexed with new index
+    # @example
+    #   dv = Daru::Vector.new [3, 2, 1], index: ['c', 'b', 'a'], type: :category
+    #   dv.reindex! ['a', 'b', 'c']
+    #   # => #<Daru::Vector(3)>
+    #   #   a   1
+    #   #   b   2
+    #   #   c   3
+    def reindex! idx
+      idx = Daru::Index.new idx unless idx.is_a? Daru::Index
+      raise ArgumentError, 'Invalid index specified' unless
+        idx.to_a.sort == index.to_a.sort
+
+      old_categories = categories
+      data = idx.map { |i| self[i] }
+      initialize_core_attributes data
+      self.categories = old_categories
+      self.index = idx
+      self
+    end
+
+    # Non-destructive version of #reorder!
+    # See #reorder!
+    def reorder order
+      dup.reorder! order
+    end
+
+    # Non-destructive version of #reindex!
+    # See #reindex!
+    def reindex index
+      dup.reindex! index
+    end
```

Comment:
Removing it.

BTW One question, I have written specs for `reindex!`, should I also write specs for `reindex`?